bftengine/src/bftengine/KeyManager.cpp

```diff
@@ -64,18 +30,40 @@ std::string KeyManager::generateCid() {
   return cid;
 }
 
-std::string KeyManager::onKeyExchange(const KeyExchangeMsg& kemsg) {
-  LOG_DEBUG(GL, "KEY EXCHANGE MANAGER  msg " << kemsg.toString());
+std::string KeyManager::onKeyExchange(KeyExchangeMsg& kemsg, const uint64_t& sn) {
```

Comment:
Reference is redundant for primitive types.
Please remove all over.

Reply:
const ref is c++ idiom, of course, that for primitives it saves nothing but idioms are important.
and if I relate to the comment below, if I remove the ref I should remove the const as well, which IMO is incorrect function signature for the purpose.

Reply:
I do not see why the "idiom" is important and how it helps here.
Changes made to a variable passed by value do not affect the original value.
P.S. I meant to remove both `const` and `&`.

Reply:
I understand what you meant, I think that removing const is wrong.
See my comment below.

Reply:
and idioms are very important, they help the code to be more concise, readable, and clear to any c++ programmer.
Any c++ programmer knows the meaning of const ref signature